src/sheet_metal_conus.py

```diff
@@ -3,188 +3,338 @@
+            - center point, start angle, end angle, and a given style.
+            unitfactor scales all dimensions.
+        """
+        attribs = {'style' : simplestyle.formatStyle(style),
+        inkex.addNS('label','inkscape') : label,
+        inkex.addNS('cx','sodipodi')     :str(centerX * unitFactor),
+        inkex.addNS('cy','sodipodi')     :str(centerY * unitFactor),
+        inkex.addNS('rx','sodipodi')     :str(radiusX * unitFactor),
+        inkex.addNS('ry','sodipodi')     :str(radiusY * unitFactor),
+        inkex.addNS('start','sodipodi')  :str(startAngle),
+        inkex.addNS('end','sodipodi')    :str(endAngle),
+        inkex.addNS('open','sodipodi')   :'true',    #all ellipse sectors we will draw are open
+        inkex.addNS('type','sodipodi')   :'arc' }
+        ell = inkex.etree.SubElement(parent, inkex.addNS('path','svg'), attribs )
+
+    def drawLine(self, (x1, y1), (x2, y2), style, label, unitFactor, parent):
+        """ Draws a line from given point 1 to given point 2 with a given style.
+        """
+        line_attribs = {'style' : simplestyle.formatStyle(style), inkex.addNS('label','inkscape') : label,
+                        'd' : 'M %s,%s L %s,%s' % (x1*unitFactor, y1*unitFactor, x2*unitFactor, y2*unitFactor)}
+        line = inkex.etree.SubElement(parent, inkex.addNS('path','svg'), line_attribs )
+
+    def beVerbose(self, dictCone, unitFactor, parent):
+        """ Verbose output of calculated values. 
+            Can be used for debugging purposes or if calculated values needed.
         """
-        inkex.debug( "Base diameter: " + str(dictCone['diaBase'] ) )
-        inkex.debug( "Cut diameter: " + str(dictCone['diaCut'] ) )
-        inkex.debug( "Cone height: " + str(dictCone['heightCone'] ) )
-        inkex.debug( "Short radius: " + str(dictCone['shortRadius']) )
-        inkex.debug( "Long radius: " + str(dictCone['longRadius']) )
-        inkex.debug( "Angle of circle sector: " + str(dictCone['angle']) + " radians (= " + str(math.degrees(dictCone['angle'])) + " degrees)")
-        inkex.debug( "Chord length of base arc: " + str(dictCone['chordBase']) )
-        inkex.debug( "Chord length of cut arc: " + str(dictCone['chordCut']) )
-        inkex.debug( "ptA: " + str(dictCone['ptA']) )
-        inkex.debug( "ptB: " + str(dictCone['ptB']) )
-        inkex.debug( "ptC: " + str(dictCone['ptC']) )
-        inkex.debug( "ptD: " + str(dictCone['ptD']) )
-        inkex.debug( "DICT: " + str(dictCone) )
+        # unpack
+        base_dia = dictCone['diaBase']
+        cut_dia = dictCone['diaCut']
+        cone_height = dictCone['heightCone']
+        shortradius = dictCone['shortRadius']
+        longradius = dictCone['longRadius']
+        angle = dictCone['angle']
+        chord_base = dictCone['chordBase']
+        chord_cut = dictCone['chordCut']
+        ptA = dictCone['ptA']
+        ptB = dictCone['ptB']
+        ptC = dictCone['ptC']
+        ptD = dictCone['ptD']
+
+        # styles for markup
```

Comment:
I see your reasoning and I can imagine that it is a lot of work.
But as far as I am concerned I would prefer the dimensioning comply to some international standard because that is how everybody is (or should be) accustomed to. 
Here are some examples: 
- http://nptel.ac.in/courses/112103019/module1/lec3/images/11.png
- http://ednotebook.hostgator.co.in/introduction
- http://www.maelabs.ucsd.edu/mae_guides/cad/dimensioning/dimensioning_fundementals.htm
- http://de.slideshare.net/anoopkrishna325/dimensioning-standards

Probably you are right that it can get crowded but this problem is usually solved by extension lines. 
On the other hand I can understand that it might be a lot of work and since your solution is better for an end-user than mine was I could live with it as it is now.
But if you see my point and can afford the time to change it I  would appreciate you to do so ....

Terminology

* Dimension lines: Lines located between extension lines ending with an arrow and to include a numerical value. They should be spaced uniformly approximately .375 to .500 inches apart.
* Extension lines: Extend away from a view to indicate a size or location constraint origin. When extension lines cross object or extension lines, no gap in either line should be made.